api/job/v1beta1/jobstate.proto

```diff
@@ -0,0 +1,56 @@
+syntax = "proto3";
+
+package api.job.v1beta1;
+
+option go_package = "github.com/openmarketplaceengine/openmarketplaceengine/internal/api/job/v1beta1";
+
+/**
+  bronx/api/src/enums/job_states.ts
+  JOB_STATES = {
+    AVAILABLE: "AVAILABLE",
+    GO_TO_PICKUP: "GO_TO_PICKUP",
+    NEAR_PICKUP: "NEAR_PICKUP",
+    AWAITING_PICKUP: "AWAITING_PICKUP",
+    ON_TRIP: "ON_TRIP",
+    NEAR_DESTINATION: "NEAR_DESTINATION",
+    AT_DESTINATION: "AT_DESTINATION",
+    COMPLETE: "COMPLETE",
+    COMPLETE_EARLY: "COMPLETE_EARLY",
+    CANCELED: "CANCELED",
+    CANCELED_BY_DRIVER: "CANCELED_BY_DRIVER",
+    CANCELED_BY_RIDER: "CANCELED_BY_RIDER",
+    RIDER_NO_SHOW: "RIDER_NO_SHOW",
+  };
+ */
+
+enum JobState {
+  JOB_STATE_UNSPECIFIED = 0;
+  JOB_STATE_AVAILABLE = 1;
+  JOB_STATE_GO_TO_PICKUP = 2;
+  JOB_STATE_NEAR_PICKUP = 3;
+  JOB_STATE_AWAITING_PICKUP = 4;
+  JOB_STATE_ON_TRIP = 5;
+  JOB_STATE_NEAR_DESTINATION = 6;
+  JOB_STATE_AT_DESTINATION = 7;
+  JOB_STATE_COMPLETE = 8;
+  JOB_STATE_COMPLETE_EARLY = 9;
+  JOB_STATE_CANCELED = 10;
+  JOB_STATE_CANCELED_BY_DRIVER = 11;
+  JOB_STATE_CANCELED_BY_RIDER = 12;
+  JOB_STATE_RIDER_NO_SHOW = 13;
+}
+
+service JobStateService {
+  rpc GetJobState(JobStateRequest) returns (JobStateResponse) {}
+  rpc SetJobState(JobStateRequest) returns (JobStateResponse) {}
```

Comment:
@kepkap @vlkolmakov 

I think we should reduce API surface area. Check out [this guide](https://cloud.google.com/apis/design/standard_methods).

Some endpoints seemingly make the same database call. For example, `GetJobState` does the same thing internally as `GetJob`. (For this kind of scenario, a [FieldMask](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf?hl=en#google.protobuf.FieldMask) can provide the client with the ability to specify which fields they want returned). Similarly, `SetJobState` can be merged in with `UpdateJob`. (A FieldMask can dictate to the server which fields should be updated).

```protobuf
 service JobService {
  rpc GetJob(GetJobRequest) returns (GetJobResponse) {}

  // Updates the job.
  // Supports updating the job's state.
  rpc UpdateJob(UpdateJobRequest) returns (UpdateJobResponse) {}

  // List jobs.
  // Supports filtering.
  rpc ListJobs(ListJobsRequest) returns (ListJobsResponse) {}
}

message ListJobsRequest {
  // if specified, we filter jobs by this state
  JobState state = 1;
}

message ListJobsResponse {
  repeated Job jobs = 1;
}
```

Reply:
if grpc supports this oob - awesome, looks very interesting. almost reaching parity with graphql :-)

Reply:
> For example, `GetJobState` does the same thing internally as `GetJob`

No. Completely different DB queries.

Reply:
A job's state isn't a column on the job table?

Reply:
We don't know upfront the final number of rpc services. If Uber operates a 28 million lines of code, we should prepare ourselves that the number of calls will only grow.

Having everything under `JobService` looks nice for now, until we add 20 more calls. I prefer we split gRPC calls per logic domain:
```
JobImpExpService
JobStateService
JobGeoService
...
```

Reply:
I think there are two distinct conversations happening here.

One is about how we model stuff in the persistence layer. (I think we should have a `job` table with a `state` column).

Another is about how we package our protobufs and draw service boundaries. I think a tollgate service might make sense, because tollgates are logically distinct from workers. However, a worker's location and a worker's state feels inherently like a part of the worker (that's why `SetWorkerState` and `UpdateWorkerLocation` should in my view just be `UpdateWorker` with a FieldMask).

Also, I generally think of a proto service as supporting CRUD. So I think it's slightly more expected to see this:
```protobuf
service FooService {
  rpc CreateFoo (CreateFooRequest) returns (CreateFooResponse) {}
  rpc GetFoo (GetFooRequest) returns (GetFooResponse) {}
  rpc ListFoo (ListFooRequest) returns (ListFooResponse) {}
  rpc UpdateFoo (UpdateFooRequest) returns (UpdateFooResponse) {}
}
```
as opposed to this:
```protobuf
service FooCreateService {
  rpc CreateFoo (CreateFooRequest) returns (CreateFooResponse) {}
}
service FooReadService {
  rpc GetFoo (GetFooRequest) returns (GetFooResponse) {}
  rpc ListFoo (ListFooRequest) returns (ListFooResponse) {}
}
// etc
```

I don't think there's a right or wrong approach here, and I can't really find any written material on when you'd opt to create a new `service`.